lib/spack/spack/cmd/list.py

```diff
@@ -40,18 +50,27 @@ def setup_parser(subparser):
         help='Optional glob patterns to filter results.')
     subparser.add_argument(
         '-s', '--sensitive', action='store_true', default=False,
-        help='Use case-sensitive filtering. Default is case sensitive, '
+        help='Use case-sensitive filtering. Default is case insensitive, '
         'unless the query contains a capital letter.')
```

Comment:
@adamjstewart @tgamblin What would you like me to do here ? Personally I would remove the option and go always case insensitive....